packages/react-core/src/components/Tabs/OverflowTab.tsx

```diff
@@ -0,0 +1,127 @@
+import { Popper } from '../../helpers';
+import { Menu, MenuContent, MenuList, MenuItem } from '../Menu';
+import { TabsContext } from './TabsContext';
+import { TabProps } from './Tab';
+
+export interface OverflowTabProps extends React.HTMLProps<HTMLLIElement> {
+  /** Additional classes added to the overflow tab */
+  className?: string;
+  /** Element to append the menu displaying the overflow tabs to */
+  menuAppendTo?: HTMLElement | (() => HTMLElement);
+  /** The tabs that should be displayed in the menu */
+  overflowingTabs?: TabProps[];
+  /** Flag which shows the count of overflowing tabs when enabled */
+  showTabCount?: boolean;
+}
+
+export const OverflowTab: React.FunctionComponent<OverflowTabProps> = ({
+  className,
+  menuAppendTo,
+  overflowingTabs = [],
+  showTabCount,
+  ...props
+}: OverflowTabProps) => {
+  const menuRef = React.useRef<HTMLDivElement>();
+  const overflowTabRef = React.useRef<HTMLLIElement>();
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+
+  const { localActiveKey, handleTabClick } = React.useContext(TabsContext);
+
+  const handleMenuKeys = (ev: KeyboardEvent) => {
+    const menuContainsEventTarget = menuRef?.current?.contains(ev.target as Node);
+    if (isExpanded && menuContainsEventTarget && ev.key === 'Escape') {
+      setIsExpanded(!isExpanded);
+      overflowTabRef?.current.focus();
+    }
+  };
+
+  const handleClick = (ev: MouseEvent) => {
+    const clickIsOutsideMenu = !menuRef?.current?.contains(ev.target as Node);
+    const clickIsOutsideOverflowTab = !overflowTabRef?.current?.contains(ev.target as Node);
+    if (isExpanded && clickIsOutsideMenu && clickIsOutsideOverflowTab) {
+      setIsExpanded(false);
+    }
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('click', handleClick);
+    window.addEventListener('keydown', handleMenuKeys);
+
+    return () => {
+      window.removeEventListener('click', handleClick);
+      window.removeEventListener('keydown', handleMenuKeys);
+    };
+  }, [isExpanded, menuRef, overflowTabRef]);
+
+  const selectedTab = overflowingTabs.find(tab => tab.eventKey === localActiveKey);
+  const tabTitle = selectedTab?.title ? selectedTab.title : 'More';
+
+  const toggleMenu = () => {
+    setIsExpanded(prevIsExpanded => !prevIsExpanded);
+    setTimeout(() => {
+      if (menuRef?.current) {
+        const firstElement = menuRef.current.querySelector('li > button,input:not(:disabled)');
+        firstElement && (firstElement as HTMLElement).focus();
+      }
+    }, 0);
+  };
+
+  const overflowTab = (
+    <li
+      className={css(styles.tabsItem, 'pf-m-overflow', selectedTab && styles.modifiers.current, className)}
+      ref={overflowTabRef}
+      {...props}
+    >
+      <button
+        className={css(styles.tabsLink, isExpanded && styles.modifiers.expanded)}
+        onClick={() => toggleMenu()}
+        aria-label="More tabs. Click this button to reveal the other tabs which you can select"
+      >
```

Comment:
It'd be worth adding the following attributes to the button toggle:

- `aria-haspopup="menu"`
-  `aria-expanded` based on the isExpanded state to better convey the button can be expanded, since you can navigate out of the overflow menu via VO without it closing and also for the following issue.


The hardcoded `aria-label` should probably be removed as well. Right now when I select a tab from the overflow menu, VO announces "More tabs..." rather than the tab I had selected. Allowing the aria-label to be set via a prop would be a good idea, though, if we expect users to omit visible text/pass in an icon to the button toggle.

Reply:
Do you think it would be better to just enable consumers to add their own labels, or to improve the automatic aria label and have it announced that the user is on a tab as well when they have a tab selected?

Or of course I could do both and just fall back to the auto-generated label when the consumer doesn't provide one.